### PR TITLE
Filling power snapshots roundNumber and blockTime

### PR DESCRIPTION
**Pull request by mohammadranjbarz**

### CODE REVIEW

File: src/entities/CronJob.ts [hunk 1]
```diff
@@ -1,28 +1,12 @@
...
-} from 'typeorm';
-import { Project, ProjStatus } from './project';
-import { Donation } from './donation';
-import { Reaction } from './reaction';
-import { AccountVerification } from './accountVerification';
-import { ProjectStatusHistory } from './projectStatusHistory';
-import { ProjectVerificationForm } from './projectVerificationForm';
-import { UserPower } from './userPower';
-import { PowerBoosting } from './powerBoosting';
-import { findPowerBoostingsCountByUserId } from '../repositories/powerBoostingRepository';
```
Comment (mohammadranjbarz) [inline src/entities/CronJob.ts:19]: There were some unused imports

File: src/services/cronJobs/syncUserPowers.ts [hunk 2]
```diff
@@ -63,13 +64,23 @@ export const getPreviousGivbackRoundInfo = (): {
...
+  fromTimestamp: number;
+  toTimestamp: number;
 } => {
   const firstGivbackRoundTimeStamp = Number(
     process.env.FIRST_GIVBACK_ROUND_TIME_STAMP,
   );
   const givbackRoundLength = Number(process.env.GIVPOWER_ROUND_DURATION);
 
-  const now = Math.floor(new Date().getTime() / 1000);
+  const now = getTimestampInSeconds(date);
```
Comment (mohammadranjbarz) [inline src/services/cronJobs/syncUserPowers.ts]: Move it to a function to make it reusable

File: src/services/cronJobs/syncUserPowers.ts [hunk 3]
```diff
@@ -63,13 +64,23 @@ export const getPreviousGivbackRoundInfo = (): {
   previousGivbackRound: number;
   fromTimestamp: number;
   toTimestamp: number;
+} => {
```
Comment (aminlatifi) [inline src/services/cronJobs/syncUserPowers.ts:68]: There is something wrong here I guess. @mohammadranjbarz can we use the same code of getPowerBoostingSnapshotRound function in src/repositories/powerBoostingRepository.ts

File: src/services/powerSnapshotServices.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+export const fillIncompletePowerSnapshots = async (): Promise<void> => {
+  const incompletePowerSnapshots = await findInCompletePowerSnapShots();
+  for (const powerSnapshot of incompletePowerSnapshots) {
+    const roundNumber = getRoundNumberByDate(
+      powerSnapshot.time,
+    ).previousGivbackRound;
+    const blockNumber = await getBlockByTime(
+      getTimestampInSeconds(powerSnapshot.time),
+    );
+    await updatePowerSnapShots({
```
Comment (aminlatifi) [inline src/services/powerSnapshotServices.ts:18]: We can just fill the entity and call save() method on that.

File: src/repositories/powerSnapshotRepository.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  id: number,
+): Promise<PowerSnapshot | undefined> => {
+  return PowerSnapshot.createQueryBuilder()
+    .where('id=:id', {
+      id,
+    })
+    .getOne();
+};
+
+export const updatePowerSnapShots = async (params: {
```
Comment (aminlatifi) [inline src/repositories/powerSnapshotRepository.ts:22]: @mohammadranjbarz if you agree with the other comment, we can remote this method

File: src/repositories/powerSnapshotRepository.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+export const findInCompletePowerSnapShots = async (): Promise<
+  PowerSnapshot[]
+> => {
+  return PowerSnapshot.createQueryBuilder()
+    .where('"blockNumber" IS NULL')
+    .getMany();
+};
+
+export const findPowerSnapshotById = async (
```
Comment (aminlatifi) [inline src/repositories/powerSnapshotRepository.ts:12]: Why we need this function and consequently its tests?

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/services/cronJobs/syncUserPowers.ts (hunk 3 above)
Comment (mohammadranjbarz) [reply src/services/cronJobs/syncUserPowers.ts:68]: Done

File: src/services/powerSnapshotServices.ts (hunk 4 above)
Comment (mohammadranjbarz) [reply src/services/powerSnapshotServices.ts:18]: I didnt move the query to service file, but  as we talk I changed that in the repository funciton

File: src/repositories/powerSnapshotRepository.ts (hunk 6 above)
Comment (mohammadranjbarz) [reply src/repositories/powerSnapshotRepository.ts:12]: I needed this query for using in test cases, so I wrote that  once in repository and reuse it in tests

**Pull request merged**